src/lib/env/_user-env.js

```diff
@@ -83,6 +83,17 @@ module.exports = function populateUserEnv (params, callback) {
       return callback(error)
     }
   }
+  else if (env) {
+    Object.entries(env).forEach(entry => {
+      let [ key, value ] = entry
+      if (typeof value === 'string') userEnv[key] = value
+      else if (typeof value === 'undefined') delete userEnv[key]
+      else {
```

Comment:
Thinking we should `toUpperCase()` all env vars here since `child_process` won't do that for us. Also observing that we should probably force-uppercase everything in `userEnv` too, probably as a last operation before passing along the `userEnv` object here: https://github.com/architect/sandbox/blob/main/src/lib/env/_user-env.js#L99 (I can do that after merge if you don't get to it.)

Reply:
@ryanblock what if we take care of the `toUpperCase()` for all `userEnv` variable names after the new "env option" has been processed. Then we eliminate any possible confusion for the developer who may be using mixed case for environment variables and is trying to replace them programmatically for a test (this is because `userEnv[key] = value` above is case-sensitive).

Reply:
If they're not continually normalized, then wouldn't that create a situation for where you couldn't reliably do things like unset env vars? For example: you have a `.env` file with `Foo=bar`, then pass `{ env: { foo: undefined } }`; unless everything is upcased, `Foo`, it'd be missed, no?